src/core/server/elasticsearch/elasticsearch_service.ts

```diff
@@ -199,7 +206,15 @@ export class ElasticsearchService
     return new ClusterClient(
       config,
       this.coreContext.logger.get('elasticsearch', type),
+      this.getAuditorFactory,
       getAuthHeaders
     );
   }
+
+  private getAuditorFactory = () => {
+    if (!this.auditorFactory) {
```

Comment:
This is not necessary for the new ES client instantiating during the `start` phase.

Reply:
I'm surprised this doesn't cause failures. That means we don't have any KP plugin performing ES calls during the `setup` phase?

Reply:
They shouldn't. It could cause data loss since data aren't migrated at this point.
This method is called from `asScoped` that should be called with `KiabanaRequest`, which can be created only when the server bound to a port.

Reply:
Oh right, that's only for scoped clients.